src/main/java/uk/gov/hmcts/reform/idam/web/strategic/PolicyService.java

```diff
@@ -130,10 +131,12 @@ private EvaluatePoliciesAction checkNoActionsBlockingUser(ResponseEntity<Evaluat
                 final boolean hasAttributes = resultItem.getAttributes() != null && resultItem.getAttributes() instanceof Map;
                 final boolean mfaRequiredAttributeIsTrue = hasAttributes && asList(ADVICE_KEY_MFA_REQUIRED_STRING_VALUE)
                     .equals(((Map) resultItem.getAttributes()).get(ADVICE_KEY_MFA_REQUIRED));
+                final boolean hasOnlyMfaRequiredAttribute = mfaRequiredAttributeIsTrue
+                    && ((Map) resultItem.getAttributes()).size() == 1;
 
                 // if mfaRequired we downgrade action to mfa_required but still need to check for other possible
                 // actions blocking the user even with mfa
-                if (mfaRequiredAttributeIsTrue) {
+                if (mfaRequiredAttributeIsTrue && hasOnlyMfaRequiredAttribute) {
```

Comment:
This is a bit confusing - isn't mfaRequiredAttributeIsTrue already part of the logic for hasOnlyMfaRequiredAttribute ?

Reply:
this is to fix a bug we found but we'll raise a ticket to re-visit this

so basically the problem today is that we cannot differentiate an mfa policy from a block user policy